---
title: "How to estimate local recombination rates with the Marey map method"
output: rmarkdown::html_vignette
bibliography: references.bib
vignette: >
  %\VignetteIndexEntry{RecombinationMap}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

library(ggplot2)
```

The package is loaded with the following command.

```{r setup}
library(EasyMareyMap)
```


# Create a MareyMap object

You can import the example dataset based on linkage maps for the five chromosomes of the plant *Arabidopsis thaliana* [@serin_construction_2017].


The datasets for the 'EasyMareyMap' package are formatted according to the specifications of input files of the MareyMap package [@rezvoy_mareymap:_2007]. It is a tab-separated file with five columns named 'set', 'map, 'mkr', 'phys', 'gen', 'vld'. Genomic coordinates are in basepairs (bp) and genetic distances in cM.

You can create a `MareyMap` object. It must contain a single map (i.e. one chromosome).


```{r importMap}
m = read.table("../inst/extdata/Arabidopsis_thaliana.txt",
               header = T,
               sep = "\t")
# Column names of the input Marey map
colnames(m)
head(m)
```


```{r mareyMapObject}
# The marey map is imported into a 'mareyMap' object
m = m[which(m$map == "1"),]
marey = mareyMap(m)
# names of slots in the 'mareyMap' object
names(marey)

# See details in ?mareyMap
colnames(marey$mareyMap)
colnames(marey$recMap)
colnames(marey$mareyCI)
```


The `EasyMareyMap` package provides many functions to directly access data or compute summary statistics directly. Moreover, to allow further development and implementation of new features, data and metadata are stored in the `mareyMap` object and easily accessible.


The `mareyMap` object is a S3 class object (see [S3 class](https://adv-r.hadley.nz/s3.html)). It contains a list of slots:
- `mareyMap` stores the input Marey map, in the format specified above.
- `recMap` stores the estimated recombination map (cM/Mb), with confidence intervals estimated by bootstrap.
- `mareyCI` stores estimated genetic positions of the same genomic windows as the recombination map, instead of recombination rates, with confidence intervals estimated by bootstrap. It can be a good practice to compare genetic positions interpolated in `mareyCI` with genetic positions observed in `mareyMap` to evaluate the quality of the interpolation method.
- then a list of attributes of the Marey map and recombination map (e.g. chromosome name and length, interpolation method...). It is important to note that the interpolation model is stored in the `model` slot and all its attributes can be accessed.






There is a function to display a `summary` of a `mareyMap` object.

```{r summary}
summary(marey)
```


# Manipulate a Marey map


The quality of the input data (the Marey map) is crucial for the performance of the Marey map interpolation function. The input data must be checked carefully before interpolation, and ambiguous markers and errors must be removed or corrected.


The `EasyMareyMap` package provides user-friendly functions to evaluate data quality and correct the most frequent errors. You can also look the `mapFuser` package for similar procedures and more features such as consensus maps [@van2017mapfuser].



Maps with a low quality often show outlying markers that will strongly negatively affect the performance of interpolation. Outlying markers are mapped outside the global trend of the Marey map. In the figure below, I add some outlier markers to the Marey map around the position 17Mb, just to learn how to remove them. There is a function to `plot` the Marey map of a `mareyMap` object. Markers kept for further interpolation are in blue. Markers discarded are in red. 


```{r plotOutliers, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "Add random outliers in the Marey map."}
marey_outliers = marey

tmp = marey_outliers$mareyMap

d = data.frame(set = unique(tmp$set),
               map = unique(tmp$map),
               mkr = "outlier",
               phys = c(17855000, 17853000, 17858000, 17840000, 17890000),
               gen = c(34, 32, 36, 33, 27),
               vld = TRUE,
               predict.se = NA)
tmp = rbind(tmp, d)
marey_outliers$mareyMap = tmp
rm(tmp)

plot(marey_outliers)
```


## Visual outlier selection

It is possible to perform outlier selection in a visual mode. You just need to draw a region around markers to remove. Left-click to place points of the polygon, then right-click to close the region and return to the script. All markers selected are automatically marked as removed (`vld =  FALSE`).

```
marey = outlierSelection(marey, "1")
```

<!-- TODO Add a photo of mkr selection -->




## Mask genomic regions


Markers can be masked (unmasked) in genomic intervals with the `mask.interval` function (`unmask.interval` respectively). Fo example, you might want to mask regions where markers are difficult to map or have higher sequencing/calling error.


```{r plot, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "Marey maps of chromosome 1 in *Arabidopsis thaliana*. Invalid markers are shown in red."}
phys = c(17855000, 17853000, 17858000, 17840000, 17890000)
gen = c(34, 32, 36, 33, 27)
pos = data.frame(phys = phys, gen = gen)
marey_outliers = mask.marker(marey_outliers, pos)

plot(marey_outliers)

rm(marey_outliers)
```




## Correction of scaffold order


The Marey map must be a monotonous increasing function. However, if some regions are not in the same order in the linkage map and the reference genome, they appear as a large segment decreasing with genomic distance. This can be due to biological reasons (structural variants, i.e. inversions) or alignment errors in the genome assembly (the scaffold has been ordered in the wrong direction). It is advised to evaluate the putative cause of this particular error before to correct it. If the cause is not well understood, we advise to discard the entire map. Otherwise, if you think it is an assembly error that can corrected, we provide a function `flip.interval()` to flip the genetic distances on this interval, in order to get them in the same order as the reference genome. Though the error is probably due to the assembly, we made the choice to flip genetic distances in order to conserve the genome annotation.

Below I flipped two intervals, simulating what is happening when some scaffolds of the reference genome and the linkage map are not in the same order. Then I use the same `flip.interval()` function to put them back in the right order.



```{r plotFlip, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "Flipping a genomic interval."}
interv = data.frame(start = c(1, 25000000), end = c(10000000, 50000000))
tmp = flip.interval(marey, intervals = interv)
plot(tmp)

tmp = flip.interval(tmp, intervals = interv)
plot(tmp)
```


You can flip the whole chromosome by specifying a single interval from one to the chromosome length.


```{r plotFlipChromosome, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "Flipping the entire chromosome"}
wholechromosome = data.frame(start = c(1), end = c(marey$chromosomeLength))
tmp = flip.interval(marey, intervals = wholechromosome)
plot(tmp)

tmp = flip.interval(tmp, intervals = wholechromosome)
plot(tmp)
```




# Estimate a recombination map

Once you have checked (and eventually corrected) the Marey map, you can estimate the recombination rates along the chromosome. Recombination rates are in cM/bp. It is the core feature of the `EasyMareyMap` package. The procedure to estimate the recombination map is done with the `recombinationMap()` function which returns a `mareyMap` object with an estimated recombination map (`recMap` slot). A description of the interpolation method is available in Brazier and Gl√©min [@brazierDiversityDeterminantsRecombination2022]. The `recombinationMap()` is designed to be user-friendly (only requires a few arguments, functions to plot and summarize), script-friendly to batch analyses (e.g. hundreds of genetic maps), highly automated (automatic fine-tuning of the smoothing parameter) and provide high reproducibility (all parameters and settings are stored in the `mareyMap` object and easily accessible).


```{r estimateMap}
res = recombinationMap(marey, method = "loess", verbose = FALSE)
```


```{r plotInterpolation, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "Marey maps of five chromosomes in *Arabidopsis thaliana* with the Marey interpolation line and a 95% C.I. (invalid markers are shown in red)."}
plot(res)
```



```{r headMap, echo=TRUE, message=FALSE}
head(res$recMap)
```


```{r summaryResults, echo=TRUE, message=FALSE}
summary(res)
```

The recombination landscape can be plotted with the `plot_recombinationMap()` function. We implemented a bootstrapped procedure (random resampling of markers) to estimate confidence intervals along the genome. It is an important feature to evaluate the quality of the recombination map.



```{r plotMap, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 8, fig.align="center", fig.cap = "Recombination landscape of *Arabidopsis thaliana*. The gray area is the 95% confidence interval around the mean estimate."}
plot_recombinationMap(res)
```



# Quality control and map validation


The validity and goodness of fit of statistical interpolation can be assessed with the linear regression of fitted values as a function of observed genetic distances. Once I fitted a loess function, I print the goodness of fit with `summary`.


```{r fitLoess, echo=TRUE, message=FALSE}
res_loess = recombinationMap(marey, method = "loess", verbose = FALSE)
summary(res_loess)
```

It can be used to compare the goodness of fit of different interpolation methods and hyperparameters (i.e. smoothing parameter)
I can compare the performance of loess and smooth splines on the same Marey map.


```{r fitSpline, echo=TRUE, message=FALSE}
res_spline = recombinationMap(marey, method = "spline", verbose = FALSE)
summary(res_spline)
```


In this case, the two interpolation methods are similarly good. It can be verified by visual comparison.


```{r fitFigure, echo=TRUE, message=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "Goodness of fit of the Marey interpolated function, either with Loess regression or smooth Splines interpolation."}
df_loess = data.frame(method = "loess",
                     y = res_loess$mareyMap$gen,
                     fitted = res_loess$fitted,
                     residuals = res_loess$residuals)
df_spline = data.frame(method = "spline",
                     y = res_spline$mareyMap$gen,
                     fitted = res_spline$fitted,
                     residuals = res_spline$residuals) 
df_compare = rbind(df_loess, df_spline)

p1 = ggplot(df_compare, aes(x = y, y = fitted)) +
  geom_point() +
  geom_smooth(method = "lm", se = TRUE) +
  facet_wrap(~ method) +
  theme_bw()
p1

p2 = ggplot(df_compare, aes(x = residuals, y = fitted)) +
  geom_point() +
  geom_smooth(method = "lm", se = TRUE) +
  facet_wrap(~ method) +
  theme_bw()
p2
```


# Changing the physical map

Genetic positions are mapped onto a reference genome (physical positions). Hence manipulating genomic intervals on this physical reference (comparing genomic intervals) or changing the mapping of genetic markers for a new reference map are important features for comparative analyses (e.g. comparing between genomic distances and gene distances as in Brazier and Gl√©min 2022).


## Estimating recombination rates for a list of genomic intervals


By default, a grid of windows of constant size is mapped along the chromosome to produce the recombination map. But you can also provide a data frame of custom start/end positions to estimate the recombination rate within specific genomic intervals or in a custom set of genomic windows along the genome.


```{r customWindows}
pos = data.frame(start = seq(20000, 40000000, by = 40000),
                 end = seq(20000, 40000000, by = 40000) + 20000)
res = recombinationMap(res, method = "loess", windows = pos, verbose = FALSE)
```

This custom positions can represent any genomic feature (e.g. gene, exon) or marker (e.g. SNP) for which you want to estimate the average recombination rate. However keep in mind that interpolation is at a large scale (i.e. does not detect fine scale variations of recombination rates) and close markers can have exactly or very similar values due to the large scale smoothing.



```{r customWindows2}
genes = data.frame(chromosome = "4",
                   start = c(60000, 99000, 123000, 267400, 500000),
                   end = c(65000, 99500, 1234000, 267401, 500001))
pos = data.frame(start = genes$start,
                 end = genes$end)
res2 = recombinationMap(res, method = "loess", windows = pos, verbose = FALSE)

res2$recMap
```



## Interpolate new genetic positions



You may want to interpolate the genetic distances for a new set of markers. It is possible to interpolate these new positions with the `predictGeneticMap()` function, passing to the `predict` function corresponding to the interpolation method (e.g. `predict.loess` for the `fit.loess` method).

It is recommended to use the same smoothing parameter previously estimated with the automatic calibration procedure.


```{r predictGeneticMap}
# Re-map on a new set of markers
new = seq(50000, 30450000, by = 1000000)
newMap = predictGeneticMap(res, new)

summary(newMap)
```


```{r newGeneticMap, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 6, fig.align="center", fig.cap = "New positions on chromosome 1, with the confidence interval (+- 1.96*se)."}
ggplot2::ggplot(data = newMap$mareyMap, aes(x = phys/10^6, y = gen)) +
  geom_point(alpha = 0.4) +
  labs(x = "Genomic position (Mb)", y = "Genetic distance (cM)") +
  geom_errorbar(aes(ymin = gen - 1.96*predict.se, ymax = gen + 1.96*predict.se)) +
  theme(axis.line = element_line(),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            panel.border = element_blank(),
            panel.background = element_blank(),
            plot.title = element_text(color="black", face="bold.italic",hjust = 0.5),
            plot.subtitle = element_text(color="black",hjust = 0.5),
            axis.title.x = element_text(color="black"),
            axis.title.y = element_text(color="black"),
            axis.text=element_text(colour="black"),
            legend.position = "right")
```


# Comparative analysis of recombination landscapes


## Chromosome-wide statistics


To  quantitatively describe and compare recombination maps, it is important to estimate summary statistics of the mean and variance of recombination rates. Many summary statistic functions of heterogeneity in recombination rates can be applied directly to the `mareyMap` object.
These statistics, estimated on the whole chromosome, can be used to compare chromosomes in a single species, but also chromosomes between populations or species.

The most simple statistics are the mean, weighted mean (weighted by the windows length), median and the variance.

```{r sumStats}
mean(res)
median(res)
weighted.mean(res)
variance(res)
```



Veller et al. (2019) proposed a rigorous measure of intra-chromosomal genetic shuffling [@veller2019rigorous]. The measure depends on the distribution of crossovers along the chromosome and total genome size. The genetic distances must be transformed back to recombinant fractions. The inverse of the mapping function used to calculate genetic distances must be applied.


```{r veller}
genome_size = 119050000
veller(res, genomeSize = genome_size, mappingFunction = "none")
veller(res, genomeSize = genome_size, mappingFunction = "haldane")
veller(res, genomeSize = genome_size, mappingFunction = "kosambi")
```


The periphery-bias ratio, as computed in Brazier and Gl√©min (2022), is a measure of the relative recombination rate in the periphery of the chromosome (default is 10%) compared to the chromosome average recombination rate. A periphery-bias ratio close to one indicate equal recombination at the periphery and the rest of the chromosome, while a ratio higher than one indicates less recombination at the periphery. Because chromosomes can be asymmetric, the measure can be calculated either on the left or right half of the chromosome, both sides or one side at random.


```{r peripherybias}
peripherybias(res, proportion_periphery = 0.1,  chromosome_arm = "random")
peripherybias(res, proportion_periphery = 0.1,  chromosome_arm = "left")
peripherybias(res, proportion_periphery = 0.1,  chromosome_arm = "right")
peripherybias(res, proportion_periphery = 0.1,  chromosome_arm = "both")
```



The coefficient of variation is a standardized measure of dispersion of a probability distribution or frequency distribution. It is defined as the ratio of the standard deviation to the mean.


```{r coefficient_variation}
coefficientVariation(res)
```


The Gini index is a measure of statistical dispersion intended to represent inequality between measures, here recombination rates.
A Gini coefficient of 0 reflects perfect equality, while a Gini coefficient of 1 reflects maximal inequality among values


```{r gini}
gini(res)
```


The Lorenz curve is a graphical representation of the heterogeneity of rates in a dataset, here recombination rates.
It plots the ordered distribution of relative recombination rates as a function of their relative physical size.
A Lorenz curve always starts at (0,0) and ends at (1,1).
The more the curve is close to the diagonal, the less heterogeneous is the distribution of recombination rates.


```{r lorenz}
lorenz(res)
```





## Compare multiple maps mapped on the same chromosome


Different genetic map can be generated in different populations, or between sexes, and mapped on the same reference genome. If the maps share the same set of markers, it is easy to plot them together. Here I compare male, female and sex-averaged recombination maps in *Homo spaiens*, a species experiencing heterochiasmy [].


```{r dataHeterochiasmy}
m = read.table("../data/Homo_sapiens_female.txt",
               header = T,
               sep = " ")
m = m[which(m$map == "Chromosome 01"),]
marey_female = mareyMap(m)
marey_female = recombinationMap(marey_female, method = "loess", verbose = FALSE)


m = read.table("../data/Homo_sapiens_male.txt",
               header = T,
               sep = " ")
m = m[which(m$map == "Chromosome 01"),]
marey_male = mareyMap(m)
marey_male = recombinationMap(marey_male, method = "loess", verbose = FALSE)


m = read.table("../data/Homo_sapiens_mean.txt",
               header = T,
               sep = " ")
m = m[which(m$map == "Chromosome 01"),]
marey_avg = mareyMap(m)
marey_avg = recombinationMap(marey_avg, method = "loess", verbose = FALSE)
```



```{r statsHeterochiasmy}
mean(marey_male)
mean(marey_female)
mean(marey_avg)

gini(marey_male)
gini(marey_female)
gini(marey_avg)

peripherybias(marey_male, proportion_periphery = 0.1,  chromosome_arm = "random")
peripherybias(marey_female, proportion_periphery = 0.1,  chromosome_arm = "random")
peripherybias(marey_avg, proportion_periphery = 0.1,  chromosome_arm = "random")
```


```{r plotMareyHeterochiasmy, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 8, fig.align="center", fig.cap = "Female, male and sex-averaged Marey maps of *Homo sapiens* chromosome 1."}
df_female = marey_female$mareyMap
df_male = marey_male$mareyMap
df_avg = marey_avg$mareyMap

df = rbind(df_female,
           df_male,
           df_avg)

  
ggplot(data = df, aes(x = phys/10^6, y = gen, colour = set, group = set)) +
      geom_point() +
      labs(x = "Genomic position (Mb)", y = "Genetic distance (cM)") +
      theme_bw()
```



```{r plotMapHeterochiasmy, message=FALSE, warning=FALSE, echo=FALSE, fig.height = 4, fig.width = 8, fig.align="center", fig.cap = "Female, male and sex-averaged recombination landscape of *Homo sapiens* chromosome 1. The shaded area is the 95% confidence interval around the mean estimate."}
df_female = marey_female$recMap
df_male = marey_male$recMap
df_avg = marey_avg$recMap

df = rbind(df_female,
           df_male,
           df_avg)

df$point = (df$start + df$end)/2

df$recRate = df$recRate * 10^6
df$upperRecRate = df$upperRecRate * 10^6
df$lowerRecRate = df$lowerRecRate * 10^6

ggplot(data = df, aes(x = point/10^6, y = recRate)) +
    geom_line(aes(group = set, color = set)) +
    geom_ribbon(aes(x = point/10^6, ymin = lowerRecRate, ymax = upperRecRate, fill = set), alpha = 0.2) +
    labs(x = "Genomic position (Mb)", y = "Recombination rate (cM/Mb)")  
```






## Compare between chromosomes/species




```{r importMapMaize}
m = read.table("../data/Zea_mays_MaizeGDBConsensus_v4.txt",
               header = T,
               sep = "\t")
m = m[which(m$map == "2"),]
maize = mareyMap(m)

maize = recombinationMap(maize, verbose = FALSE)

summary(maize)
```


In order to compare maps between two species with contrasted recombination landscapes, for example Arabidopsis and maize, visual methods like the Lorenz curve or expressing physical distances in relative distances cam be very efficient. If you need to compare quantitatively a large number of maps, summary statistics like the Gini index or the periphery-bias ratio are efficient measure of the heterogeneity of recombination rates along the chromosome.



In order to get the data frame instead of the Lorenz plot, you can specify `return.plot = FALSE`. It works also for the `plot.mareyMap` and `plot_recombination` functions.


```{r lorenzCompare, message=FALSE, warning=FALSE, echo=TRUE, fig.height = 4, fig.width = 8, fig.align="center", fig.cap = "Comparative Lorenz curves between Arabidopsis chromosome 1 and maize chromosome 1."}
lorenz_arabido = lorenz(res, return.plot = FALSE)
lorenz_maize = lorenz(maize, return.plot = FALSE)

df =  rbind(lorenz_arabido, lorenz_maize)

diagonal = data.frame(x = seq(0, 1, by = 0.01),
                        y = seq(0, 1, by = 0.01))
ggplot(data = df, aes(x = relativePhys, y = relativeGen, color = set)) +
    geom_line() +
    geom_line(data = diagonal, aes(x = x, y = y), color = "Grey") +
    xlim(0, 1) +
    ylim(0, 1) +
    xlab("Proportion of genomic distance") +
    ylab("Proportion of genetic distance") +
    theme_bw()
```



```{r relativeCompare, message=FALSE, warning=FALSE, echo=TRUE, fig.height = 4, fig.width = 8, fig.align="center", fig.cap = "Comparative recombination landscapes between Arabidopsis chromosome 1 and maize chromosome 1. Genomic distances are relative distances scaled by the total chromosome size."}
relative_arabido = res$recMap
relative_maize = maize$recMap

relative_arabido$relative_start = relative_arabido$start / res$chromosomeLength
relative_arabido$relative_end = relative_arabido$end / res$chromosomeLength
relative_arabido$relative_recRate = relative_arabido$recRate / max(relative_arabido$recRate, na.rm = TRUE)
relative_arabido$relative_lowerRecRate = relative_arabido$lowerRecRate / max(relative_arabido$lowerRecRate, na.rm = TRUE)
relative_arabido$relative_upperRecRate = relative_arabido$upperRecRate / max(relative_arabido$upperRecRate, na.rm = TRUE)


relative_maize$relative_start = relative_maize$start / maize$chromosomeLength
relative_maize$relative_end = relative_maize$end / maize$chromosomeLength
relative_maize$relative_recRate = relative_maize$recRate / max(relative_maize$recRate, na.rm = TRUE)
relative_maize$relative_lowerRecRate = relative_maize$lowerRecRate / max(relative_maize$lowerRecRate, na.rm = TRUE)
relative_maize$relative_upperRecRate = relative_maize$upperRecRate / max(relative_maize$upperRecRate, na.rm = TRUE)


df = rbind(relative_arabido,
           relative_maize)

df$point = (df$relative_start + df$relative_end)/2


ggplot(data = df, aes(x = point, y = recRate)) +
    geom_line(aes(group = set, color = set)) +
    geom_ribbon(aes(x = point, ymin = lowerRecRate, ymax = upperRecRate, fill = set), alpha = 0.2) +
    labs(x = "Relative genomic position", y = "Recombination rate (cM/Mb)") 


ggplot(data = df, aes(x = point, y = relative_recRate)) +
    geom_line(aes(group = set, color = set)) +
    geom_ribbon(aes(x = point, ymin = relative_lowerRecRate, ymax = relative_upperRecRate, fill = set), alpha = 0.2) +
    labs(x = "Relative genomic position", y = "Relative recombination rate") 
```







# References
