<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>How to estimate local recombination rates with the Marey map method • EasyMareyMap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="How to estimate local recombination rates with the Marey map method">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EasyMareyMap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/RecombinationMap.html">How to estimate local recombination rates with the Marey map method</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>How to estimate local recombination rates with
the Marey map method</h1>
            
      

      <div class="hidden name"><code>RecombinationMap.Rmd</code></div>

    </div>

    
    
<p>The <code>EasyMareyMap</code> package for R has been implemented to
estimate recombination maps from genetic maps (or linkage maps) using
the Marey map approach <span class="citation">(Rezvoy et al.
2007)</span>. The original MareyMap package was designed for interactive
graphical use only and does not provide scripting capabilities. It does
not allow to process many maps at a time. See the original MareyMap
package here <a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btm315" class="external-link">Rezvoy
et al. 2007</a> and <a href="https://CRAN.R-project.org/package=MareyMap" class="external-link">Siberchicot et
al. 2017</a>.</p>
<p>Our <code>EasyMareyMap</code> package is designed to provide easy
access to the data (stored in slots within objects) and the functions
via the command line, hence a more machine-friendly approach, in order
to maximize scriptability, reproducibility and scalability to large
datasets. It offers a statistical framework to estimate recombination
maps with the automatic calibration of hyperparameters and a bootstrap
procedure to measure the uncertainty of our estimates. The method has
been used and validated in <span class="citation">(Brazier and Glémin
2022)</span>.</p>
<p>Moreover, it offers a statistical framework to estimate recombination
maps with the automatic calibration of hyperparameters and a bootstrap
procedure to measure the uncertainty of our estimates, which is an
advantage for robust data analyses. The method has been used and
validated in <span class="citation">(Brazier and Glémin 2022)</span>.
Our <code>EasyMareyMap</code> package adds these new features: *
automatic calibration of the smoothing parameter * bootstraps to
estimate a 95% confidence interval * graphical zone selection to exclude
a bunch of outlier markers * a graphical and statistical framework for
comparative genomics (compare Marey maps between species, populations or
genomes)</p>
<p>The package is loaded with the following command.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EasyMareyMap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>Two S3 objects are at the core of the package: * the
<code>marey_map</code> object contains all the data and metadata
associated with a single map. It contains slots with the Marey map, the
recombination map (after estimating it), and additional metadata, such
as model goodness of fit. * the <code>comparative_marey_map</code>
object is a collection of <code>marey_map</code> objects over which you
can apply <code>EasyMareyMap</code> functions or iterate with your
custom function. Hence it is a convenient object for manipulating,
computing and comparing mutliple maps (i.e. multiple chromosomes and/or
datasets).</p>
<p><code>EasyMareyMap</code> functions allow you to: * estimate the
recombination map from the Marey map, using loess or smooth spline
regression * estimate a bootstrapped confidence interval for
recombination rates * predict genetic positions for a new set of markers
* plot one or many Marey maps, plot one or many recombination maps *
compute chromosome wide summary statistics (mean, median, variance, gini
index, periphery-bias ratio…) * comparative plots (lorenz curve, broken
stick, see below for details) * apply some corrections to a raw Marey
map (e.g. outlier removal)</p>
<div class="section level2">
<h2 id="create-a-marey_map-object">Create a <code>marey_map</code> object<a class="anchor" aria-label="anchor" href="#create-a-marey_map-object"></a>
</h2>
<p>You can import the example dataset based on linkage maps for the five
chromosomes of the plant <em>Arabidopsis thaliana</em> <span class="citation">(Serin et al. 2017)</span>. The datasets for the
‘EasyMareyMap’ package are formatted according to the specifications of
input files of the MareyMap package <span class="citation">(Rezvoy et
al. 2007)</span>. It is a tab-separated file with five columns named
‘set’, ‘map, ’mkr’, ‘phys’, ‘gen’, ‘vld’. Genomic coordinates are in
basepairs (bp) and genetic distances in cM.</p>
<p>You can then create a <code>marey_map</code> object. It must contain
a single map (i.e. one chromosome). We will see later how to treat
multiple maps in a convenient way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"../inst/extdata/Arabidopsis_thaliana.txt"</span>,</span>
<span>               header <span class="op">=</span> <span class="cn">T</span>,</span>
<span>               sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="co"># Column names of the input Marey map</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "set"  "map"  "mkr"  "phys" "gen"  "vld"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span>
<span><span class="co">#&gt;                              set map        mkr   phys  gen  vld</span></span>
<span><span class="co">#&gt; 1 Arabidopsis_thaliana_Serin2017   1 RSM_1_0.05  50000 0.00 TRUE</span></span>
<span><span class="co">#&gt; 2 Arabidopsis_thaliana_Serin2017   1 RSM_1_0.15 150000 0.31 TRUE</span></span>
<span><span class="co">#&gt; 3 Arabidopsis_thaliana_Serin2017   1 RSM_1_0.25 250000 0.63 TRUE</span></span>
<span><span class="co">#&gt; 4 Arabidopsis_thaliana_Serin2017   1 RSM_1_0.35 350000 1.28 TRUE</span></span>
<span><span class="co">#&gt; 5 Arabidopsis_thaliana_Serin2017   1 RSM_1_0.45 450000 1.92 TRUE</span></span>
<span><span class="co">#&gt; 6 Arabidopsis_thaliana_Serin2017   1 RSM_1_0.55 550000 2.24 TRUE</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The marey map is imported into a 'marey_map' object</span></span>
<span><span class="va">m</span> <span class="op">=</span> <span class="va">m</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">m</span><span class="op">$</span><span class="va">map</span> <span class="op">==</span> <span class="st">"1"</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">marey</span> <span class="op">=</span> <span class="fu"><a href="../reference/marey_map.html">marey_map</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span>
<span><span class="co"># names of slots in the 'marey_map' object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">marey</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "mareyMap"               "recMap"                 "mareyCI"               </span></span>
<span><span class="co">#&gt;  [4] "chromosomeName"         "chromosomeLength"       "linkageMapLength"      </span></span>
<span><span class="co">#&gt;  [7] "interpolationMethod"    "model"                  "fitted"                </span></span>
<span><span class="co">#&gt; [10] "residuals"              "goodness.r.squared"     "goodness.adj.r.squared"</span></span>
<span><span class="co">#&gt; [13] "goodness.p"             "windows"                "smoothingParam"        </span></span>
<span><span class="co">#&gt; [16] "nBootstrap"</span></span>
<span></span>
<span><span class="co"># See details in ?marey_map</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">marey</span><span class="op">$</span><span class="va">marey_map</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">marey</span><span class="op">$</span><span class="va">recMap</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "set"          "map"          "start"        "end"          "recRate"     </span></span>
<span><span class="co">#&gt; [6] "lowerRecRate" "upperRecRate"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">marey</span><span class="op">$</span><span class="va">mareyCI</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "set"                    "map"                    "physicalPosition"      </span></span>
<span><span class="co">#&gt; [4] "geneticPositioncM"      "upperGeneticPositioncM" "lowerGeneticPositioncM"</span></span></code></pre></div>
<p>The <code>EasyMareyMap</code> package provides many functions to
directly access data or compute summary statistics directly. Moreover,
to allow further development and implementation of new features, data
and metadata are stored in the <code>marey_map</code> object and easily
accessible.</p>
<p>The <code>marey_map</code> object is a S3 class object (see <a href="https://adv-r.hadley.nz/s3.html" class="external-link">S3 class</a>). It contains a list
of slots: - <code>mareyMap</code> stores the input Marey map, in the
format specified above. - <code>recMap</code> stores the estimated
recombination map (cM/Mb), with confidence intervals estimated by
bootstrap. - <code>mareyCI</code> stores estimated genetic positions of
the same genomic windows as the recombination map, instead of
recombination rates, with confidence intervals estimated by bootstrap.
It can be a good practice to compare genetic positions interpolated in
<code>mareyCI</code> with genetic positions observed in
<code>mareyMap</code> to evaluate the quality of the interpolation
method. - then a list of attributes of the Marey map and recombination
map (e.g. chromosome name and length, interpolation method…). It is
important to note that the interpolation model is stored in the
<code>model</code> slot and all its attributes can be accessed.</p>
<p>There is a function to display a <code>summary</code> of a
<code>marey_map</code> object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">marey</span><span class="op">)</span></span>
<span><span class="co">#&gt; ===================== Summary of the Marey map ==================</span></span>
<span><span class="co">#&gt;  Dataset:  Arabidopsis_thaliana_Serin2017 </span></span>
<span><span class="co">#&gt;  Chromosome name:  1 </span></span>
<span><span class="co">#&gt;  Chromosome length (Mb):  30.45 </span></span>
<span><span class="co">#&gt;  Linkage map length (cM):  117 </span></span>
<span><span class="co">#&gt;  Number of markers:  275 </span></span>
<span><span class="co">#&gt;  Markers density (marker/Mb):  9.031199 </span></span>
<span><span class="co">#&gt;  =========================== Recombination map ===================</span></span>
<span><span class="co">#&gt;  Recombination rates (cM/Mb):</span></span>
<span><span class="co">#&gt;  Min.   1st Qu.    Median      Mean   3rd Qu.      Max.</span></span>
<span><span class="co">#&gt; ======================== Interpolation ==========================</span></span>
<span><span class="co">#&gt; Interpolation method:   </span></span>
<span><span class="co">#&gt;  Smoothing parameter:   </span></span>
<span><span class="co">#&gt;  Number of windows:  0 </span></span>
<span><span class="co">#&gt;  Number of bootstraps:   </span></span>
<span><span class="co">#&gt; ================= Goodness of fit (fitted ~ y) ===================</span></span>
<span><span class="co">#&gt;  fitted ~ y R squared:   </span></span>
<span><span class="co">#&gt;  fitted ~ y adjusted R squared:   </span></span>
<span><span class="co">#&gt;  fitted ~ y p-value   </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="manipulate-a-marey-map">Manipulate a Marey map<a class="anchor" aria-label="anchor" href="#manipulate-a-marey-map"></a>
</h2>
<p>The quality of the input data (the Marey map) is crucial for the
performance of the Marey map interpolation function. The input data must
be checked carefully before interpolation, and ambiguous markers and
errors must be removed or corrected.</p>
<p>The <code>EasyMareyMap</code> package provides user-friendly
functions to evaluate data quality and correct the most frequent errors.
You can also look the <code>mapFuser</code> package for similar
procedures and more features such as consensus maps <span class="citation">(Muijen et al. 2017)</span>.</p>
<p>Maps with a low quality often show outlying markers that will
strongly negatively affect the performance of interpolation. Outlying
markers are mapped outside the global trend of the Marey map. In the
figure below, I add some outlier markers to the Marey map around the
position 17Mb, just to learn how to remove them. There is a function to
<code>plot</code> the Marey map of a <code>marey_map</code> object.
Markers kept for further interpolation are in blue. Markers discarded
are in red.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">marey_outliers</span> <span class="op">=</span> <span class="va">marey</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="va">marey_outliers</span><span class="op">$</span><span class="va">mareyMap</span></span>
<span></span>
<span><span class="va">d</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>set <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">$</span><span class="va">set</span><span class="op">)</span>,</span>
<span>               map <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">$</span><span class="va">map</span><span class="op">)</span>,</span>
<span>               mkr <span class="op">=</span> <span class="st">"outlier"</span>,</span>
<span>               phys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">17855000</span>, <span class="fl">17853000</span>, <span class="fl">17858000</span>, <span class="fl">17840000</span>, <span class="fl">17890000</span><span class="op">)</span>,</span>
<span>               gen <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">34</span>, <span class="fl">32</span>, <span class="fl">36</span>, <span class="fl">33</span>, <span class="fl">27</span><span class="op">)</span>,</span>
<span>               vld <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               predict.se <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="va">d</span><span class="op">)</span></span>
<span><span class="va">marey_outliers</span><span class="op">$</span><span class="va">mareyMap</span> <span class="op">=</span> <span class="va">tmp</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">marey_outliers</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotOutliers-1.png" alt="Add random outliers in the Marey map." width="576"><p class="caption">
Add random outliers in the Marey map.
</p>
</div>
<div class="section level3">
<h3 id="visual-outlier-selection">Visual outlier selection<a class="anchor" aria-label="anchor" href="#visual-outlier-selection"></a>
</h3>
<p>It is possible to perform outlier selection in a visual mode. You
just need to draw a region around markers to remove. Left-click to place
points of the polygon, then right-click to close the region and return
to the script. All markers selected are automatically marked as removed
(<code>vld =  FALSE</code>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">marey</span> <span class="op">=</span> <span class="fu"><a href="../reference/outlier_selection.html">outlier_selection</a></span><span class="op">(</span><span class="va">marey</span><span class="op">)</span></span></code></pre></div>
<!-- TODO Add a photo of mkr selection -->
</div>
<div class="section level3">
<h3 id="mask-genomic-regions">Mask genomic regions<a class="anchor" aria-label="anchor" href="#mask-genomic-regions"></a>
</h3>
<p>Markers can be masked (unmasked) in genomic intervals with the
<code>mask.interval</code> function (<code>unmask.interval</code>
respectively). Fo example, you might want to mask regions where markers
are difficult to map or have higher sequencing/calling error.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">phys</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">17855000</span>, <span class="fl">17853000</span>, <span class="fl">17858000</span>, <span class="fl">17840000</span>, <span class="fl">17890000</span><span class="op">)</span></span>
<span><span class="va">gen</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">34</span>, <span class="fl">32</span>, <span class="fl">36</span>, <span class="fl">33</span>, <span class="fl">27</span><span class="op">)</span></span>
<span><span class="va">pos</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>phys <span class="op">=</span> <span class="va">phys</span>, gen <span class="op">=</span> <span class="va">gen</span><span class="op">)</span></span>
<span><span class="va">marey_outliers</span> <span class="op">=</span> <span class="fu"><a href="../reference/mask.marker.html">mask.marker</a></span><span class="op">(</span><span class="va">marey_outliers</span>, <span class="va">pos</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">marey_outliers</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plot-1.png" alt="Marey maps of chromosome 1 in *Arabidopsis thaliana*. Invalid markers are shown in red." width="576"><p class="caption">
Marey maps of chromosome 1 in <em>Arabidopsis thaliana</em>. Invalid
markers are shown in red.
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">marey_outliers</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="correction-of-scaffold-order">Correction of scaffold order<a class="anchor" aria-label="anchor" href="#correction-of-scaffold-order"></a>
</h3>
<p>The Marey map must be a monotonous increasing function. However, if
some regions are not in the same order in the linkage map and the
reference genome, they appear as a large segment decreasing with genomic
distance. This can be due to biological reasons (structural variants,
i.e. inversions) or alignment errors in the genome assembly (the
scaffold has been ordered in the wrong direction). It is advised to
evaluate the putative cause of this particular error before to correct
it. If the cause is not well understood, we advise to discard the entire
map. Otherwise, if you think it is an assembly error that can corrected,
we provide a function <code><a href="../reference/flip.interval.html">flip.interval()</a></code> to flip the genetic
distances on this interval, in order to get them in the same order as
the reference genome. Though the error is probably due to the assembly,
we made the choice to flip genetic distances in order to conserve the
genome annotation.</p>
<p>Below I flipped two intervals, simulating what is happening when some
scaffolds of the reference genome and the linkage map are not in the
same order. Then I use the same <code><a href="../reference/flip.interval.html">flip.interval()</a></code> function to
put them back in the right order.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">interv</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">25000000</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10000000</span>, <span class="fl">50000000</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="fu"><a href="../reference/flip.interval.html">flip.interval</a></span><span class="op">(</span><span class="va">marey</span>, intervals <span class="op">=</span> <span class="va">interv</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotFlip-1.png" alt="Flipping a genomic interval." width="576"><p class="caption">
Flipping a genomic interval.
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="fu"><a href="../reference/flip.interval.html">flip.interval</a></span><span class="op">(</span><span class="va">tmp</span>, intervals <span class="op">=</span> <span class="va">interv</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotFlip-2.png" alt="Flipping a genomic interval." width="576"><p class="caption">
Flipping a genomic interval.
</p>
</div>
<p>You can flip the whole chromosome by specifying a single interval
from one to the chromosome length.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wholechromosome</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">marey</span><span class="op">$</span><span class="va">chromosomeLength</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="fu"><a href="../reference/flip.interval.html">flip.interval</a></span><span class="op">(</span><span class="va">marey</span>, intervals <span class="op">=</span> <span class="va">wholechromosome</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotFlipChromosome-1.png" alt="Flipping the entire chromosome" width="576"><p class="caption">
Flipping the entire chromosome
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="fu"><a href="../reference/flip.interval.html">flip.interval</a></span><span class="op">(</span><span class="va">tmp</span>, intervals <span class="op">=</span> <span class="va">wholechromosome</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotFlipChromosome-2.png" alt="Flipping the entire chromosome" width="576"><p class="caption">
Flipping the entire chromosome
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="estimate-a-recombination-map">Estimate a recombination map<a class="anchor" aria-label="anchor" href="#estimate-a-recombination-map"></a>
</h2>
<p>Once you have checked (and eventually corrected) the Marey map, you
can estimate the recombination rates along the chromosome. Recombination
rates are in cM/bp. It is the core feature of the
<code>EasyMareyMap</code> package. The procedure to estimate the
recombination map is done with the <code><a href="../reference/recombination_map.html">recombination_map()</a></code>
function which returns a <code>marey_map</code> object with an estimated
recombination map (<code>recMap</code> slot). A description of the
interpolation method is available in Brazier and Glémin <span class="citation">(Brazier and Glémin 2022)</span>. The
<code><a href="../reference/recombination_map.html">recombination_map()</a></code> is designed to be user-friendly (only
requires a few arguments, functions to plot and summarize),
script-friendly to batch analyses (e.g. hundreds of genetic maps),
highly automated (automatic fine-tuning of the smoothing parameter) and
provide high reproducibility (all parameters and settings are stored in
the <code>marey_map</code> object and easily accessible).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/recombination_map.html">recombination_map</a></span><span class="op">(</span><span class="va">marey</span>, method <span class="op">=</span> <span class="st">"loess"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotInterpolation-1.png" alt="Marey maps of five chromosomes in *Arabidopsis thaliana* with the Marey interpolation line and a 95% C.I. (invalid markers are shown in red)." width="576"><p class="caption">
Marey maps of five chromosomes in <em>Arabidopsis thaliana</em> with the
Marey interpolation line and a 95% C.I. (invalid markers are shown in
red).
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">recMap</span><span class="op">)</span></span>
<span><span class="co">#&gt;                              set map start    end      recRate upperRecRate</span></span>
<span><span class="co">#&gt; 1 Arabidopsis_thaliana_Serin2017   1 0e+00  99999          NaN           NA</span></span>
<span><span class="co">#&gt; 2 Arabidopsis_thaliana_Serin2017   1 1e+05 199999 4.302678e-06 4.789104e-06</span></span>
<span><span class="co">#&gt; 3 Arabidopsis_thaliana_Serin2017   1 2e+05 299999 4.364607e-06 4.877508e-06</span></span>
<span><span class="co">#&gt; 4 Arabidopsis_thaliana_Serin2017   1 3e+05 399999 4.387298e-06 4.887265e-06</span></span>
<span><span class="co">#&gt; 5 Arabidopsis_thaliana_Serin2017   1 4e+05 499999 4.394586e-06 4.860013e-06</span></span>
<span><span class="co">#&gt; 6 Arabidopsis_thaliana_Serin2017   1 5e+05 599999 4.396032e-06 4.842377e-06</span></span>
<span><span class="co">#&gt;   lowerRecRate</span></span>
<span><span class="co">#&gt; 1           NA</span></span>
<span><span class="co">#&gt; 2 3.663111e-06</span></span>
<span><span class="co">#&gt; 3 3.764711e-06</span></span>
<span><span class="co">#&gt; 4 3.824480e-06</span></span>
<span><span class="co">#&gt; 5 3.891871e-06</span></span>
<span><span class="co">#&gt; 6 3.960459e-06</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; ===================== Summary of the Marey map ==================</span></span>
<span><span class="co">#&gt;  Dataset:  Arabidopsis_thaliana_Serin2017 </span></span>
<span><span class="co">#&gt;  Chromosome name:  1 </span></span>
<span><span class="co">#&gt;  Chromosome length (Mb):  30.45 </span></span>
<span><span class="co">#&gt;  Linkage map length (cM):  117 </span></span>
<span><span class="co">#&gt;  Number of markers:  275 </span></span>
<span><span class="co">#&gt;  Markers density (marker/Mb):  9.031199 </span></span>
<span><span class="co">#&gt;  =========================== Recombination map ===================</span></span>
<span><span class="co">#&gt;  Recombination rates (cM/Mb):</span></span>
<span><span class="co">#&gt;  Min.   1st Qu.    Median      Mean   3rd Qu.      Max.</span></span>
<span><span class="co">#&gt; 1e+06 0.477 3.4 4.08 3.91 4.8 6.24 </span></span>
<span><span class="co">#&gt; ======================== Interpolation ==========================</span></span>
<span><span class="co">#&gt; Interpolation method:  loess </span></span>
<span><span class="co">#&gt;  Smoothing parameter:  0.2 </span></span>
<span><span class="co">#&gt;  Number of windows:  304 </span></span>
<span><span class="co">#&gt;  Number of bootstraps:  1000 </span></span>
<span><span class="co">#&gt; ================= Goodness of fit (fitted ~ y) ===================</span></span>
<span><span class="co">#&gt;  fitted ~ y R squared:  0.9925402 </span></span>
<span><span class="co">#&gt;  fitted ~ y adjusted R squared:  0.9925129 </span></span>
<span><span class="co">#&gt;  fitted ~ y p-value  2.05262e-292 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span></code></pre></div>
<p>The recombination landscape can be plotted with the
<code><a href="../reference/plot_recombination_map.html">plot_recombination_map()</a></code> function. We implemented a
bootstrapped procedure (random resampling of markers) to estimate
confidence intervals along the genome. It is an important feature to
evaluate the quality of the recombination map.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_recombination_map.html">plot_recombination_map</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMap-1.png" alt="Recombination landscape of *Arabidopsis thaliana*. The gray area is the 95% confidence interval around the mean estimate." width="768"><p class="caption">
Recombination landscape of <em>Arabidopsis thaliana</em>. The gray area
is the 95% confidence interval around the mean estimate.
</p>
</div>
</div>
<div class="section level2">
<h2 id="quality-control-and-map-validation">Quality control and map validation<a class="anchor" aria-label="anchor" href="#quality-control-and-map-validation"></a>
</h2>
<p>The validity and goodness of fit of statistical interpolation can be
assessed with the linear regression of fitted values as a function of
observed genetic distances. Once I fitted a loess function, I print the
goodness of fit with <code>summary</code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_loess</span> <span class="op">=</span> <span class="fu"><a href="../reference/recombination_map.html">recombination_map</a></span><span class="op">(</span><span class="va">marey</span>, method <span class="op">=</span> <span class="st">"loess"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res_loess</span><span class="op">)</span></span>
<span><span class="co">#&gt; ===================== Summary of the Marey map ==================</span></span>
<span><span class="co">#&gt;  Dataset:  Arabidopsis_thaliana_Serin2017 </span></span>
<span><span class="co">#&gt;  Chromosome name:  1 </span></span>
<span><span class="co">#&gt;  Chromosome length (Mb):  30.45 </span></span>
<span><span class="co">#&gt;  Linkage map length (cM):  117 </span></span>
<span><span class="co">#&gt;  Number of markers:  275 </span></span>
<span><span class="co">#&gt;  Markers density (marker/Mb):  9.031199 </span></span>
<span><span class="co">#&gt;  =========================== Recombination map ===================</span></span>
<span><span class="co">#&gt;  Recombination rates (cM/Mb):</span></span>
<span><span class="co">#&gt;  Min.   1st Qu.    Median      Mean   3rd Qu.      Max.</span></span>
<span><span class="co">#&gt; 1e+06 0.477 3.4 4.08 3.91 4.8 6.24 </span></span>
<span><span class="co">#&gt; ======================== Interpolation ==========================</span></span>
<span><span class="co">#&gt; Interpolation method:  loess </span></span>
<span><span class="co">#&gt;  Smoothing parameter:  0.2 </span></span>
<span><span class="co">#&gt;  Number of windows:  304 </span></span>
<span><span class="co">#&gt;  Number of bootstraps:  1000 </span></span>
<span><span class="co">#&gt; ================= Goodness of fit (fitted ~ y) ===================</span></span>
<span><span class="co">#&gt;  fitted ~ y R squared:  0.9925402 </span></span>
<span><span class="co">#&gt;  fitted ~ y adjusted R squared:  0.9925129 </span></span>
<span><span class="co">#&gt;  fitted ~ y p-value  2.05262e-292 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span></code></pre></div>
<p>It can be used to compare the goodness of fit of different
interpolation methods and hyperparameters (i.e. smoothing parameter) I
can compare the performance of loess and smooth splines on the same
Marey map.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_spline</span> <span class="op">=</span> <span class="fu"><a href="../reference/recombination_map.html">recombination_map</a></span><span class="op">(</span><span class="va">marey</span>, method <span class="op">=</span> <span class="st">"spline"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res_spline</span><span class="op">)</span></span>
<span><span class="co">#&gt; ===================== Summary of the Marey map ==================</span></span>
<span><span class="co">#&gt;  Dataset:  Arabidopsis_thaliana_Serin2017 </span></span>
<span><span class="co">#&gt;  Chromosome name:  1 </span></span>
<span><span class="co">#&gt;  Chromosome length (Mb):  30.45 </span></span>
<span><span class="co">#&gt;  Linkage map length (cM):  117 </span></span>
<span><span class="co">#&gt;  Number of markers:  275 </span></span>
<span><span class="co">#&gt;  Markers density (marker/Mb):  9.031199 </span></span>
<span><span class="co">#&gt;  =========================== Recombination map ===================</span></span>
<span><span class="co">#&gt;  Recombination rates (cM/Mb):</span></span>
<span><span class="co">#&gt;  Min.   1st Qu.    Median      Mean   3rd Qu.      Max.</span></span>
<span><span class="co">#&gt; 1e+06 1.59 3.14 4.01 3.93 4.91 5.67 </span></span>
<span><span class="co">#&gt; ======================== Interpolation ==========================</span></span>
<span><span class="co">#&gt; Interpolation method:  spline </span></span>
<span><span class="co">#&gt;  Smoothing parameter:  0.2 </span></span>
<span><span class="co">#&gt;  Number of windows:  304 </span></span>
<span><span class="co">#&gt;  Number of bootstraps:  1000 </span></span>
<span><span class="co">#&gt; ================= Goodness of fit (fitted ~ y) ===================</span></span>
<span><span class="co">#&gt;  fitted ~ y R squared:  0.993156 </span></span>
<span><span class="co">#&gt;  fitted ~ y adjusted R squared:  0.9931309 </span></span>
<span><span class="co">#&gt;  fitted ~ y p-value  1.602351e-297 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span></code></pre></div>
<p>In this case, the two interpolation methods are similarly good. It
can be verified by visual comparison.</p>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/fitFigure-1.png" alt="Goodness of fit of two Marey interpolated functions, either with Loess regression or Smooth Splines interpolation. *Arabidopsis thaliana* chromosome 1." width="576"><p class="caption">
Goodness of fit of two Marey interpolated functions, either with Loess
regression or Smooth Splines interpolation. <em>Arabidopsis
thaliana</em> chromosome 1.
</p>
</div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/fitFigure-2.png" alt="Goodness of fit of two Marey interpolated functions, either with Loess regression or Smooth Splines interpolation. *Arabidopsis thaliana* chromosome 1." width="576"><p class="caption">
Goodness of fit of two Marey interpolated functions, either with Loess
regression or Smooth Splines interpolation. <em>Arabidopsis
thaliana</em> chromosome 1.
</p>
</div>
</div>
<div class="section level2">
<h2 id="changing-the-physical-map">Changing the physical map<a class="anchor" aria-label="anchor" href="#changing-the-physical-map"></a>
</h2>
<p>Genetic positions are mapped onto a reference genome (physical
positions). Hence manipulating genomic intervals on this physical
reference (comparing genomic intervals) or changing the mapping of
genetic markers for a new reference map are important features for
comparative analyses (e.g. comparing between genomic distances and gene
distances as in Brazier and Glémin 2022).</p>
<div class="section level3">
<h3 id="estimating-recombination-rates-for-a-list-of-genomic-intervals">Estimating recombination rates for a list of genomic intervals<a class="anchor" aria-label="anchor" href="#estimating-recombination-rates-for-a-list-of-genomic-intervals"></a>
</h3>
<p>By default, a grid of windows of constant size is mapped along the
chromosome to produce the recombination map. But you can also provide a
data frame of custom start/end positions to estimate the recombination
rate within specific genomic intervals or in a custom set of genomic
windows along the genome.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pos</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20000</span>, <span class="fl">40000000</span>, by <span class="op">=</span> <span class="fl">40000</span><span class="op">)</span>,</span>
<span>                 end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20000</span>, <span class="fl">40000000</span>, by <span class="op">=</span> <span class="fl">40000</span><span class="op">)</span> <span class="op">+</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/recombination_map.html">recombination_map</a></span><span class="op">(</span><span class="va">res</span>, method <span class="op">=</span> <span class="st">"loess"</span>, windows <span class="op">=</span> <span class="va">pos</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>This custom positions can represent any genomic feature (e.g. gene,
exon) or marker (e.g. SNP) for which you want to estimate the average
recombination rate. However keep in mind that interpolation is at a
large scale (i.e. does not detect fine scale variations of recombination
rates) and close markers can have exactly or very similar values due to
the large scale smoothing.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chromosome <span class="op">=</span> <span class="st">"4"</span>,</span>
<span>                   start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">60000</span>, <span class="fl">99000</span>, <span class="fl">123000</span>, <span class="fl">267400</span>, <span class="fl">500000</span><span class="op">)</span>,</span>
<span>                   end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">65000</span>, <span class="fl">99500</span>, <span class="fl">1234000</span>, <span class="fl">267401</span>, <span class="fl">500001</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pos</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">start</span>,</span>
<span>                 end <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">end</span><span class="op">)</span></span>
<span><span class="va">res2</span> <span class="op">=</span> <span class="fu"><a href="../reference/recombination_map.html">recombination_map</a></span><span class="op">(</span><span class="va">res</span>, method <span class="op">=</span> <span class="st">"loess"</span>, windows <span class="op">=</span> <span class="va">pos</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res2</span><span class="op">$</span><span class="va">recMap</span></span>
<span><span class="co">#&gt;                              set map  start     end      recRate upperRecRate</span></span>
<span><span class="co">#&gt; 1 Arabidopsis_thaliana_Serin2017   1  60000   65000 4.293182e-06 4.776110e-06</span></span>
<span><span class="co">#&gt; 2 Arabidopsis_thaliana_Serin2017   1  99000   99500 4.291914e-06 4.766776e-06</span></span>
<span><span class="co">#&gt; 3 Arabidopsis_thaliana_Serin2017   1 123000 1234000 4.306163e-06 4.627640e-06</span></span>
<span><span class="co">#&gt; 4 Arabidopsis_thaliana_Serin2017   1 267400  267401 4.375945e-06 4.937747e-06</span></span>
<span><span class="co">#&gt; 5 Arabidopsis_thaliana_Serin2017   1 500000  500001 4.388549e-06 4.907002e-06</span></span>
<span><span class="co">#&gt;   lowerRecRate</span></span>
<span><span class="co">#&gt; 1 3.527333e-06</span></span>
<span><span class="co">#&gt; 2 3.545424e-06</span></span>
<span><span class="co">#&gt; 3 3.912515e-06</span></span>
<span><span class="co">#&gt; 4 3.719777e-06</span></span>
<span><span class="co">#&gt; 5 3.847568e-06</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="interpolate-new-genetic-positions">Interpolate new genetic positions<a class="anchor" aria-label="anchor" href="#interpolate-new-genetic-positions"></a>
</h3>
<p>You may want to interpolate the genetic distances for a new set of
markers. It is possible to interpolate these new positions with the
<code><a href="../reference/predict_genetic_map.html">predict_genetic_map()</a></code> function, passing to the
<code>predict</code> function corresponding to the interpolation method
(e.g. <code>predict.loess</code> for the <code>fit.loess</code>
method).</p>
<p>It is recommended to use the same smoothing parameter previously
estimated with the automatic calibration procedure.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Re-map on a new set of markers</span></span>
<span><span class="va">new</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">50000</span>, <span class="fl">30450000</span>, by <span class="op">=</span> <span class="fl">1000000</span><span class="op">)</span></span>
<span><span class="va">new_map</span> <span class="op">=</span> <span class="fu"><a href="../reference/predict_genetic_map.html">predict_genetic_map</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">new</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">new_map</span><span class="op">)</span></span>
<span><span class="co">#&gt; ===================== Summary of the Marey map ==================</span></span>
<span><span class="co">#&gt;  Dataset:  Arabidopsis_thaliana_Serin2017 </span></span>
<span><span class="co">#&gt;  Chromosome name:  1 </span></span>
<span><span class="co">#&gt;  Chromosome length (Mb):  30.45 </span></span>
<span><span class="co">#&gt;  Linkage map length (cM):  117 </span></span>
<span><span class="co">#&gt;  Number of markers:  31 </span></span>
<span><span class="co">#&gt;  Markers density (marker/Mb):  1.018062 </span></span>
<span><span class="co">#&gt;  =========================== Recombination map ===================</span></span>
<span><span class="co">#&gt;  Recombination rates (cM/Mb):</span></span>
<span><span class="co">#&gt;  Min.   1st Qu.    Median      Mean   3rd Qu.      Max.</span></span>
<span><span class="co">#&gt; 1e+06 0.485 3.41 4.06 3.91 4.8 6.24 </span></span>
<span><span class="co">#&gt; ======================== Interpolation ==========================</span></span>
<span><span class="co">#&gt; Interpolation method:  loess </span></span>
<span><span class="co">#&gt;  Smoothing parameter:  0.2 </span></span>
<span><span class="co">#&gt;  Number of windows:  1000 </span></span>
<span><span class="co">#&gt;  Number of bootstraps:  1000 </span></span>
<span><span class="co">#&gt; ================= Goodness of fit (fitted ~ y) ===================</span></span>
<span><span class="co">#&gt;  fitted ~ y R squared:  0.9925402 </span></span>
<span><span class="co">#&gt;  fitted ~ y adjusted R squared:  0.9925129 </span></span>
<span><span class="co">#&gt;  fitted ~ y p-value  2.05262e-292 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/newGeneticMap-1.png" alt="New genetic positions (cM) on chromosome 1, with the confidence interval (+- 1.96*se) of the prediction." width="576"><p class="caption">
New genetic positions (cM) on chromosome 1, with the confidence interval
(+- 1.96*se) of the prediction.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="comparative-analysis-of-recombination-landscapes">Comparative analysis of recombination landscapes<a class="anchor" aria-label="anchor" href="#comparative-analysis-of-recombination-landscapes"></a>
</h2>
<div class="section level3">
<h3 id="chromosome-wide-statistics">Chromosome-wide statistics<a class="anchor" aria-label="anchor" href="#chromosome-wide-statistics"></a>
</h3>
<p>To quantitatively describe and compare recombination maps, it is
important to estimate summary statistics of the mean and variance of
recombination rates. Many summary statistics of heterogeneity in
recombination rates can be calculated directly with the
<code>marey_map</code> object. These statistics, estimated on the whole
chromosome, can be used to compare chromosomes in a single species, but
also chromosomes between populations or species.</p>
<p>The most simple statistics are the mean, weighted mean (weighted by
the windows length), median and the variance.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 3.912888e-06</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4.075139e-06</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html" class="external-link">weighted.mean</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 3.912888e-06</span></span>
<span><span class="fu"><a href="../reference/variance.html">variance</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1.726989e-12</span></span></code></pre></div>
<p>Veller et al. (2019) proposed a rigorous measure of intra-chromosomal
genetic shuffling <span class="citation">(Veller, Kleckner, and Nowak
2019)</span>. The measure depends on the distribution of crossovers
along the chromosome and total genome size. The genetic distances must
be transformed back to recombinant fractions. The inverse of the mapping
function used to calculate genetic distances must be applied.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genome_size</span> <span class="op">=</span> <span class="fl">119050000</span></span>
<span><span class="fu"><a href="../reference/veller.html">veller</a></span><span class="op">(</span><span class="va">res</span>, genome_size <span class="op">=</span> <span class="va">genome_size</span>, mapping_function <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Converting Morgans to recombinant fraction</span></span>
<span><span class="co">#&gt; [1] 1.394364e-06</span></span>
<span><span class="fu"><a href="../reference/veller.html">veller</a></span><span class="op">(</span><span class="va">res</span>, genome_size <span class="op">=</span> <span class="va">genome_size</span>, mapping_function <span class="op">=</span> <span class="st">"haldane"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Reversed Haldane, converting cM to recombinant fraction</span></span>
<span><span class="co">#&gt; [1] 8.453857e-07</span></span>
<span><span class="fu"><a href="../reference/veller.html">veller</a></span><span class="op">(</span><span class="va">res</span>, genome_size <span class="op">=</span> <span class="va">genome_size</span>, mapping_function <span class="op">=</span> <span class="st">"kosambi"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Reversed Kosambi, converting cM to recombinant fraction</span></span>
<span><span class="co">#&gt; [1] 9.88544e-07</span></span></code></pre></div>
<p>The periphery-bias ratio, as computed in Brazier and Glémin (2022),
is a measure of the relative recombination rate at the periphery of the
chromosome (default is 10%) compared to the chromosome average
recombination rate. A periphery-bias ratio close to one indicate equal
recombination at the periphery and the rest of the chromosome, while a
ratio higher than one indicates less recombination at the periphery.
Because chromosomes can be asymmetric, the measure can be calculated
either on the left or right half of the chromosome, both sides or one
side at random.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/peripherybias.html">peripherybias</a></span><span class="op">(</span><span class="va">res</span>, proportion_periphery <span class="op">=</span> <span class="fl">0.1</span>,  chromosome_arm <span class="op">=</span> <span class="st">"random"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1.049341</span></span>
<span><span class="fu"><a href="../reference/peripherybias.html">peripherybias</a></span><span class="op">(</span><span class="va">res</span>, proportion_periphery <span class="op">=</span> <span class="fl">0.1</span>,  chromosome_arm <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1.049341</span></span>
<span><span class="fu"><a href="../reference/peripherybias.html">peripherybias</a></span><span class="op">(</span><span class="va">res</span>, proportion_periphery <span class="op">=</span> <span class="fl">0.1</span>,  chromosome_arm <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1.320047</span></span>
<span><span class="fu"><a href="../reference/peripherybias.html">peripherybias</a></span><span class="op">(</span><span class="va">res</span>, proportion_periphery <span class="op">=</span> <span class="fl">0.1</span>,  chromosome_arm <span class="op">=</span> <span class="st">"both"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1.182814</span></span></code></pre></div>
<p>The coefficient of variation is a standardized measure of dispersion
of a probability distribution or frequency distribution. It is defined
as the ratio of the standard deviation divided by the mean.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/coefficient_variation.html">coefficient_variation</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.3358516</span></span></code></pre></div>
<p>The Gini index is a measure of statistical dispersion intended to
represent inequality between measures, here recombination rates. A Gini
coefficient of 0 reflects perfect equality, while a Gini coefficient of
1 reflects maximal inequality among values</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/gini.html">gini</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.1852593</span></span></code></pre></div>
<p>The Lorenz curve is a graphical representation of the heterogeneity
of rates in a dataset, here recombination rates. It plots the ordered
distribution of relative recombination rates as a function of their
relative physical size. A Lorenz curve always starts at (0,0) and ends
at (1,1). The more the curve is close to the diagonal, the less
heterogeneous is the distribution of recombination rates.</p>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/lorenz-1.png" alt="The Lorenz curve of *Arabidopsis thaliana* chromosome 1." width="576"><p class="caption">
The Lorenz curve of <em>Arabidopsis thaliana</em> chromosome 1.
</p>
</div>
</div>
<div class="section level3">
<h3 id="compare-multiple-maps-mapped-on-the-same-chromosome">Compare multiple maps mapped on the same chromosome<a class="anchor" aria-label="anchor" href="#compare-multiple-maps-mapped-on-the-same-chromosome"></a>
</h3>
<p>Different genetic map can be generated in different populations, or
between sexes, and mapped on the same reference genome. If the maps
share the same set of markers, it is easy to plot them together. Here I
compare male, female and sex-averaged recombination maps in <em>Homo
sapiens</em>, a species with heterochiasmy (i.e. recombination rates are
different between sexes).</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">human</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../inst/extdata/Homo_sapiens_comparative.Rds"</span><span class="op">)</span></span></code></pre></div>
<p>It is the time to present the <code>comparative_marey_map</code>
object. It is a meta-object storing multiple maps (i.e. multiple
chromosomes and multiple datasets). It allows to perform operations on a
list of maps (e.g. estimating recombination rates), compute summary
statistics (e.g. mean, median recombination rates or the gini index) and
plot these maps in the same figure in a comparative manner. The
<code>EasyMareyMap</code> package has also functions to easily
manipulate <code>comparative_marey_map</code> objects.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A short summary of the data stored</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="co">#&gt; ============== Summary of the comparative marey map ==============</span></span>
<span><span class="co">#&gt;  Datasets:  Homo sapiens (female) Homo sapiens (male) Homo sapiens (mean) ... </span></span>
<span><span class="co">#&gt;  Number of maps:  66 </span></span>
<span><span class="co">#&gt;  Mean linkage map length (cM):  163.6061 </span></span>
<span><span class="co">#&gt;  Mean chromosome size (Mb):  129.7833 </span></span>
<span><span class="co">#&gt;  Mean number of markers:  526.2727 </span></span>
<span><span class="co">#&gt;  Mean marker density (marker/Mb):  4.157088 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span>
<span></span>
<span><span class="co"># The slots in the comparative_marey_map object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "data" "set"  "map"</span></span>
<span></span>
<span><span class="co"># The class</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "comparative_marey_map"</span></span></code></pre></div>
<p>The recombination maps can be estimated in batch for the whole
comparative dataset with the function
<code><a href="../reference/comparative_recombination_maps.html">comparative_recombination_maps()</a></code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">human</span> <span class="op">=</span> <span class="fu"><a href="../reference/comparative_recombination_maps.html">comparative_recombination_maps</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span></span></code></pre></div>
<p>There are functions to easily get the Marey maps or the recombination
maps from a <code>comparative_marey_map</code> object.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/comparative_marey_to_dataframe.html">comparative_marey_to_dataframe</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     set           map       mkr    phys gen  vld predict.se</span></span>
<span><span class="co">#&gt; 1 Homo sapiens (female) Chromosome 01   D1S2217  837022   0 TRUE         NA</span></span>
<span><span class="co">#&gt; 2 Homo sapiens (female) Chromosome 01  WIAF-768  919830   0 TRUE         NA</span></span>
<span><span class="co">#&gt; 3 Homo sapiens (female) Chromosome 01 WIAF-3885  989675   0 TRUE         NA</span></span>
<span><span class="co">#&gt; 4 Homo sapiens (female) Chromosome 01  WIAF-984 2128800   3 TRUE         NA</span></span>
<span><span class="co">#&gt; 5 Homo sapiens (female) Chromosome 01    D1S243 2171436   3 TRUE         NA</span></span>
<span><span class="co">#&gt; 6 Homo sapiens (female) Chromosome 01    D1S468 3608114   8 TRUE         NA</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/comparative_interpolation_to_dataframe.html">comparative_interpolation_to_dataframe</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     set           map physicalPosition geneticPositioncM</span></span>
<span><span class="co">#&gt; 1 Homo sapiens (female) Chromosome 01            50000                 0</span></span>
<span><span class="co">#&gt; 2 Homo sapiens (female) Chromosome 01           150000                 0</span></span>
<span><span class="co">#&gt; 3 Homo sapiens (female) Chromosome 01           250000                 0</span></span>
<span><span class="co">#&gt; 4 Homo sapiens (female) Chromosome 01           350000                 0</span></span>
<span><span class="co">#&gt; 5 Homo sapiens (female) Chromosome 01           450000                 0</span></span>
<span><span class="co">#&gt; 6 Homo sapiens (female) Chromosome 01           550000                 0</span></span>
<span><span class="co">#&gt;   upperGeneticPositioncM lowerGeneticPositioncM</span></span>
<span><span class="co">#&gt; 1                      0                      0</span></span>
<span><span class="co">#&gt; 2                      0                      0</span></span>
<span><span class="co">#&gt; 3                      0                      0</span></span>
<span><span class="co">#&gt; 4                      0                      0</span></span>
<span><span class="co">#&gt; 5                      0                      0</span></span>
<span><span class="co">#&gt; 6                      0                      0</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/comparative_recmap_to_dataframe.html">comparative_recmap_to_dataframe</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     set           map start    end recRate upperRecRate</span></span>
<span><span class="co">#&gt; 1 Homo sapiens (female) Chromosome 01 0e+00  99999     NaN           NA</span></span>
<span><span class="co">#&gt; 2 Homo sapiens (female) Chromosome 01 1e+05 199999     NaN           NA</span></span>
<span><span class="co">#&gt; 3 Homo sapiens (female) Chromosome 01 2e+05 299999     NaN           NA</span></span>
<span><span class="co">#&gt; 4 Homo sapiens (female) Chromosome 01 3e+05 399999     NaN           NA</span></span>
<span><span class="co">#&gt; 5 Homo sapiens (female) Chromosome 01 4e+05 499999     NaN           NA</span></span>
<span><span class="co">#&gt; 6 Homo sapiens (female) Chromosome 01 5e+05 599999     NaN           NA</span></span>
<span><span class="co">#&gt;   lowerRecRate</span></span>
<span><span class="co">#&gt; 1           NA</span></span>
<span><span class="co">#&gt; 2           NA</span></span>
<span><span class="co">#&gt; 3           NA</span></span>
<span><span class="co">#&gt; 4           NA</span></span>
<span><span class="co">#&gt; 5           NA</span></span>
<span><span class="co">#&gt; 6           NA</span></span></code></pre></div>
<p>Let’s work with only two chromosomes and the three datasets to show
how to subset maps in a <code>comparative_marey_map</code> object. Each
map in the <code>comparative_marey_map</code> object is a single
<code>marey_map</code> object. hence you can iterate over each element
of <code>comparative_marey_map$data</code> to perform custom operations
on <code>marey_map</code> objects. There is also a function
<code><a href="../reference/merge_comparative_marey.html">merge_comparative_marey()</a></code> to merge
<code>comparative_marey_map</code> objects.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Make a logical vector of maps to keep</span></span>
<span><span class="va">subs</span> <span class="op">=</span> <span class="op">(</span><span class="va">human</span><span class="op">$</span><span class="va">map</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chromosome 02"</span> ,<span class="st">"Chromosome 05"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">human_subset</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_comparative_marey.html">subset_comparative_marey</a></span><span class="op">(</span><span class="va">human</span>, subset <span class="op">=</span> <span class="va">subs</span><span class="op">)</span></span></code></pre></div>
<p>And now compute three summary statistics on each map.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compute_stats_marey.html">compute_stats_marey</a></span><span class="op">(</span><span class="va">human_subset</span>, statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mean"</span>, <span class="st">"gini"</span>, <span class="st">"peripherybias"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $set</span></span>
<span><span class="co">#&gt; [1] "Homo sapiens (female)" "Homo sapiens (male)"   "Homo sapiens (mean)"  </span></span>
<span><span class="co">#&gt; [4] "Homo sapiens (female)" "Homo sapiens (male)"   "Homo sapiens (mean)"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $map</span></span>
<span><span class="co">#&gt; [1] "Chromosome 02" "Chromosome 02" "Chromosome 02" "Chromosome 05"</span></span>
<span><span class="co">#&gt; [5] "Chromosome 05" "Chromosome 05"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $mean</span></span>
<span><span class="co">#&gt; [1] 1.422049e-06 7.770710e-07 1.094594e-06 1.517336e-06 8.400675e-07</span></span>
<span><span class="co">#&gt; [6] 1.171480e-06</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $gini</span></span>
<span><span class="co">#&gt; [1] 0.2087580 0.3230733 0.2365540 0.2369744 0.4088750 0.2748154</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $peripherybias</span></span>
<span><span class="co">#&gt; [1] 1.531973 2.152255 1.802875 1.812777 1.985314 1.877357</span></span></code></pre></div>
<p>It is easy to compute statistics on a large set of maps (22
chromosomes in three datasets).</p>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/compareHeterochiasmy-1.png" alt="Mean recombinaiton rate per chromosome (cM/Mb) and Gini index for the female, male and sex-averaged genetic maps in human." width="768"><p class="caption">
Mean recombinaiton rate per chromosome (cM/Mb) and Gini index for the
female, male and sex-averaged genetic maps in human.
</p>
</div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/compareHeterochiasmy-2.png" alt="Mean recombinaiton rate per chromosome (cM/Mb) and Gini index for the female, male and sex-averaged genetic maps in human." width="768"><p class="caption">
Mean recombinaiton rate per chromosome (cM/Mb) and Gini index for the
female, male and sex-averaged genetic maps in human.
</p>
</div>
<p>You can plot either the Marey maps or the recombination maps with the
<code><a href="../reference/plot_comparative_marey.html">plot_comparative_marey()</a></code> or the
<code><a href="../reference/plot_comparative_recmap.html">plot_comparative_recmap()</a></code> functions. They return a ggplot
object that can be stored or customized. They allow a grouping factor
for the plot. Maps can be grouped by <code>set</code>, <code>map</code>
or both (i.e. <code>set + map</code>).</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_comparative_marey.html">plot_comparative_marey</a></span><span class="op">(</span><span class="va">human_subset</span>, group <span class="op">=</span> <span class="st">"set"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMareyHeterochiasmy-1.png" alt="Female, male and sex-averaged Marey maps of *Homo sapiens* chromosomes 2 and 5." width="768"><p class="caption">
Female, male and sex-averaged Marey maps of <em>Homo sapiens</em>
chromosomes 2 and 5.
</p>
</div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_comparative_marey.html">plot_comparative_marey</a></span><span class="op">(</span><span class="va">human_subset</span>, group <span class="op">=</span> <span class="st">"map"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMareyHeterochiasmy-2.png" alt="Female, male and sex-averaged Marey maps of *Homo sapiens* chromosomes 2 and 5." width="768"><p class="caption">
Female, male and sex-averaged Marey maps of <em>Homo sapiens</em>
chromosomes 2 and 5.
</p>
</div>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_comparative_marey.html">plot_comparative_marey</a></span><span class="op">(</span><span class="va">human_subset</span>, group <span class="op">=</span> <span class="st">"set + map"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMareyHeterochiasmy-3.png" alt="Female, male and sex-averaged Marey maps of *Homo sapiens* chromosomes 2 and 5." width="768"><p class="caption">
Female, male and sex-averaged Marey maps of <em>Homo sapiens</em>
chromosomes 2 and 5.
</p>
</div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_comparative_recmap.html">plot_comparative_recmap</a></span><span class="op">(</span><span class="va">human_subset</span>, group <span class="op">=</span> <span class="st">"set"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMapHeterochiasmy-1.png" alt="Female, male and sex-averaged recombination landscape of *Homo sapiens* chromosome 2 and 5. The shaded area is the 95% confidence interval around the mean estimate." width="768"><p class="caption">
Female, male and sex-averaged recombination landscape of <em>Homo
sapiens</em> chromosome 2 and 5. The shaded area is the 95% confidence
interval around the mean estimate.
</p>
</div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_comparative_recmap.html">plot_comparative_recmap</a></span><span class="op">(</span><span class="va">human_subset</span>, group <span class="op">=</span> <span class="st">"map"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMapHeterochiasmy-2.png" alt="Female, male and sex-averaged recombination landscape of *Homo sapiens* chromosome 2 and 5. The shaded area is the 95% confidence interval around the mean estimate." width="768"><p class="caption">
Female, male and sex-averaged recombination landscape of <em>Homo
sapiens</em> chromosome 2 and 5. The shaded area is the 95% confidence
interval around the mean estimate.
</p>
</div>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_comparative_recmap.html">plot_comparative_recmap</a></span><span class="op">(</span><span class="va">human_subset</span>, group <span class="op">=</span> <span class="st">"set + map"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/plotMapHeterochiasmy-3.png" alt="Female, male and sex-averaged recombination landscape of *Homo sapiens* chromosome 2 and 5. The shaded area is the 95% confidence interval around the mean estimate." width="768"><p class="caption">
Female, male and sex-averaged recombination landscape of <em>Homo
sapiens</em> chromosome 2 and 5. The shaded area is the 95% confidence
interval around the mean estimate.
</p>
</div>
</div>
<div class="section level3">
<h3 id="compare-between-chromosomesspecies">Compare between chromosomes/species<a class="anchor" aria-label="anchor" href="#compare-between-chromosomesspecies"></a>
</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maize</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../inst/extdata/Zea_mays_MaizeGDBConsensus_v4.Rds"</span><span class="op">)</span></span>
<span><span class="va">arabidopsis</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../inst/extdata/Arabidopsis_thaliana.Rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">maize</span><span class="op">)</span></span>
<span><span class="co">#&gt; ============== Summary of the comparative marey map ==============</span></span>
<span><span class="co">#&gt;  Datasets:  Zea_mays_MaizeGDBConsensus_v4 Zea_mays_MaizeGDBConsensus_v4 Zea_mays_MaizeGDBConsensus_v4 ... </span></span>
<span><span class="co">#&gt;  Number of maps:  10 </span></span>
<span><span class="co">#&gt;  Mean linkage map length (cM):  177.8 </span></span>
<span><span class="co">#&gt;  Mean chromosome size (Mb):  210.287 </span></span>
<span><span class="co">#&gt;  Mean number of markers:  629.9 </span></span>
<span><span class="co">#&gt;  Mean marker density (marker/Mb):  3.006639 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">arabidopsis</span><span class="op">)</span></span>
<span><span class="co">#&gt; ============== Summary of the comparative marey map ==============</span></span>
<span><span class="co">#&gt;  Datasets:  Arabidopsis_thaliana_Serin2017 Arabidopsis_thaliana_Serin2017 Arabidopsis_thaliana_Serin2017 ... </span></span>
<span><span class="co">#&gt;  Number of maps:  5 </span></span>
<span><span class="co">#&gt;  Mean linkage map length (cM):  93.8 </span></span>
<span><span class="co">#&gt;  Mean chromosome size (Mb):  23.81 </span></span>
<span><span class="co">#&gt;  Mean number of markers:  211.8 </span></span>
<span><span class="co">#&gt;  Mean marker density (marker/Mb):  8.872908 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span>
<span></span>
<span><span class="va">compar</span> <span class="op">=</span> <span class="fu"><a href="../reference/merge_comparative_marey.html">merge_comparative_marey</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">arabidopsis</span>, <span class="va">maize</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">compar</span><span class="op">)</span></span>
<span><span class="co">#&gt; ============== Summary of the comparative marey map ==============</span></span>
<span><span class="co">#&gt;  Datasets:  Arabidopsis_thaliana_Serin2017 Arabidopsis_thaliana_Serin2017 Arabidopsis_thaliana_Serin2017 ... </span></span>
<span><span class="co">#&gt;  Number of maps:  15 </span></span>
<span><span class="co">#&gt;  Mean linkage map length (cM):  149.8 </span></span>
<span><span class="co">#&gt;  Mean chromosome size (Mb):  148.128 </span></span>
<span><span class="co">#&gt;  Mean number of markers:  490.5333 </span></span>
<span><span class="co">#&gt;  Mean marker density (marker/Mb):  4.962062 </span></span>
<span><span class="co">#&gt;  ==================================================================</span></span></code></pre></div>
<p>In order to compare maps between two species with contrasted
recombination landscapes, for example <em>Arabidopsis thaliana</em> and
maize, visual methods like the Lorenz curve or expressing physical
distances in relative distances cam be very efficient. If you need to
compare quantitatively a large number of maps, summary statistics like
the Gini index or the periphery-bias ratio are efficient measures of the
heterogeneity of recombination rates along the chromosome.</p>
<p>In order to get the data frame instead of the Lorenz plot, you can
specify <code>return.plot = FALSE</code>. It works also for the
<code><a href="../reference/plot.marey_map.html">plot.marey_map()</a></code> and <code>plot_recombination</code>
functions.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/lorenz.html">lorenz</a></span><span class="op">(</span><span class="va">compar</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/lorenzCompare-1.png" alt="Comparative Lorenz curves between *Arabidopsis thaliana* and maize." width="768"><p class="caption">
Comparative Lorenz curves between <em>Arabidopsis thaliana</em> and
maize.
</p>
</div>
<p>The Broken stick model, as described in Brazier and Glémin (2022), is
also an appropriate representation to compare the distribution of
recombination rates along the genome across a large set of Marey
maps.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/brokenstick.html">brokenstick</a></span><span class="op">(</span><span class="va">compar</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="RecombinationMap_files/figure-html/brokenStick-1.png" alt="The broken stick model applied to *Arabidopsis thaliana* and maize." width="768"><p class="caption">
The broken stick model applied to <em>Arabidopsis thaliana</em> and
maize.
</p>
</div>
<p>Moreover, to compare between species, it can be interesting to use
relative genomic positions and relative recombination rates, instead of
absolute ones.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">relative_arabido</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">recMap</span></span>
<span><span class="va">relative_maize</span> <span class="op">=</span> <span class="va">maize</span><span class="op">$</span><span class="va">recMap</span></span>
<span></span>
<span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">relative_start</span> <span class="op">=</span> <span class="va">relative_arabido</span><span class="op">$</span><span class="va">start</span> <span class="op">/</span> <span class="va">res</span><span class="op">$</span><span class="va">chromosomeLength</span></span>
<span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">relative_end</span> <span class="op">=</span> <span class="va">relative_arabido</span><span class="op">$</span><span class="va">end</span> <span class="op">/</span> <span class="va">res</span><span class="op">$</span><span class="va">chromosomeLength</span></span>
<span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">relative_recRate</span> <span class="op">=</span> <span class="va">relative_arabido</span><span class="op">$</span><span class="va">recRate</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">recRate</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">relative_lowerRecRate</span> <span class="op">=</span> <span class="va">relative_arabido</span><span class="op">$</span><span class="va">lowerRecRate</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">lowerRecRate</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">relative_upperRecRate</span> <span class="op">=</span> <span class="va">relative_arabido</span><span class="op">$</span><span class="va">upperRecRate</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">relative_arabido</span><span class="op">$</span><span class="va">upperRecRate</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">relative_maize</span><span class="op">$</span><span class="va">relative_start</span> <span class="op">=</span> <span class="va">relative_maize</span><span class="op">$</span><span class="va">start</span> <span class="op">/</span> <span class="va">maize</span><span class="op">$</span><span class="va">chromosomeLength</span></span>
<span><span class="va">relative_maize</span><span class="op">$</span><span class="va">relative_end</span> <span class="op">=</span> <span class="va">relative_maize</span><span class="op">$</span><span class="va">end</span> <span class="op">/</span> <span class="va">maize</span><span class="op">$</span><span class="va">chromosomeLength</span></span>
<span><span class="va">relative_maize</span><span class="op">$</span><span class="va">relative_recRate</span> <span class="op">=</span> <span class="va">relative_maize</span><span class="op">$</span><span class="va">recRate</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">relative_maize</span><span class="op">$</span><span class="va">recRate</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">relative_maize</span><span class="op">$</span><span class="va">relative_lowerRecRate</span> <span class="op">=</span> <span class="va">relative_maize</span><span class="op">$</span><span class="va">lowerRecRate</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">relative_maize</span><span class="op">$</span><span class="va">lowerRecRate</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">relative_maize</span><span class="op">$</span><span class="va">relative_upperRecRate</span> <span class="op">=</span> <span class="va">relative_maize</span><span class="op">$</span><span class="va">upperRecRate</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">relative_maize</span><span class="op">$</span><span class="va">upperRecRate</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">relative_arabido</span>,</span>
<span>           <span class="va">relative_maize</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">point</span> <span class="op">=</span> <span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">relative_start</span> <span class="op">+</span> <span class="va">df</span><span class="op">$</span><span class="va">relative_end</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">point</span>, y <span class="op">=</span> <span class="va">recRate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">set</span>, color <span class="op">=</span> <span class="va">set</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_ribbon.html" class="external-link">geom_ribbon</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">point</span>, ymin <span class="op">=</span> <span class="va">lowerRecRate</span>, ymax <span class="op">=</span> <span class="va">upperRecRate</span>, fill <span class="op">=</span> <span class="va">set</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Relative genomic position"</span>, y <span class="op">=</span> <span class="st">"Recombination rate (cM/Mb)"</span><span class="op">)</span> </span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">point</span>, y <span class="op">=</span> <span class="va">relative_recRate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">set</span>, color <span class="op">=</span> <span class="va">set</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_ribbon.html" class="external-link">geom_ribbon</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">point</span>, ymin <span class="op">=</span> <span class="va">relative_lowerRecRate</span>, ymax <span class="op">=</span> <span class="va">relative_upperRecRate</span>, fill <span class="op">=</span> <span class="va">set</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Relative genomic position"</span>, y <span class="op">=</span> <span class="st">"Relative recombination rate"</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-brazierDiversityDeterminantsRecombination2022" class="csl-entry">
Brazier, Thomas, and Sylvain Glémin. 2022. <span>“Diversity and
Determinants of Recombination Landscapes in Flowering Plants.”</span>
Edited by Ian R. Henderson. <em>PLOS Genetics</em> 18 (8): e1010141. <a href="https://doi.org/10.1371/journal.pgen.1010141" class="external-link">https://doi.org/10.1371/journal.pgen.1010141</a>.
</div>
<div id="ref-van2017mapfuser" class="csl-entry">
Muijen, Dennis van, Ram K Basnet, Nathalie NJ Dek, Chris Maliepaard, and
Evert W Gutteling. 2017. <span>“Mapfuser: An Integrative Toolbox for
Consensus Map Construction and Marey Maps.”</span> <em>BioRxiv</em>,
200311.
</div>
<div id="ref-rezvoy_mareymap:_2007" class="csl-entry">
Rezvoy, C., D. Charif, L. Gueguen, and G. A. B. Marais. 2007.
<span>“<span>MareyMap</span>: An <span>R</span>-Based Tool with
Graphical Interface for Estimating Recombination Rates.”</span>
<em>Bioinformatics</em> 23 (16): 2188–89. <a href="https://doi.org/10.1093/bioinformatics/btm315" class="external-link">https://doi.org/10.1093/bioinformatics/btm315</a>.
</div>
<div id="ref-serin_construction_2017" class="csl-entry">
Serin, Elise A. R., L. B. Snoek, Harm Nijveen, Leo A. J. Willems, Jose
M. Jiménez-Gómez, Henk W. M. Hilhorst, and Wilco Ligterink. 2017.
<span>“Construction of a <span>High</span>-<span>Density</span>
<span>Genetic</span> <span>Map</span> from
<span>RNA</span>-<span>Seq</span> <span>Data</span> for an
<span>Arabidopsis</span> <span>Bay</span>-0 × <span>Shahdara</span>
<span>RIL</span> <span>Population</span>.”</span> <em>Frontiers in
Genetics</em> 8 (December): 201. <a href="https://doi.org/10.3389/fgene.2017.00201" class="external-link">https://doi.org/10.3389/fgene.2017.00201</a>.
</div>
<div id="ref-veller2019rigorous" class="csl-entry">
Veller, Carl, Nancy Kleckner, and Martin A Nowak. 2019. <span>“A
Rigorous Measure of Genome-Wide Genetic Shuffling That Takes into
Account Crossover Positions and Mendel’s Second Law.”</span>
<em>Proceedings of the National Academy of Sciences</em> 116 (5):
1659–68.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Thomas Brazier.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
